### PR TITLE
Fs config validation

### PR DESCRIPTION
**Pull request by phlax**
Ensure all fs configs are validated when set

### CODE REVIEW

File: tests/pootle_fs/utils.py [hunk 1]
```diff
@@ -167,10 +167,10 @@ def test_project_fs_instance_bad(english):
     project.config["pootle_fs.fs_type"] = "foo"
     with pytest.raises(NotConfiguredError):
         FSPlugin(project)
-    project.config["pootle_fs.fs_type"] = None
-    project.config["pootle_fs.fs_url"] = "bar"
+    project.config["pootle_fs.fs_type"] = "localfs"
     with pytest.raises(NotConfiguredError):
         FSPlugin(project)
+    project.config["pootle_fs.fs_url"] = "/bar"
```
Comment (unho) [inline tests/pootle_fs/utils.py]: Nit: Shouldn't the `fs_url` be set after setting `fs_type`?

File: pootle/apps/pootle_fs/getters.py [hunk 2]
```diff
@@ -59,3 +64,16 @@ def fs_url_validator_getter(**kwargs_):
...
 def fs_translation_mapping_validator_getter(**kwargs_):
     return TranslationMappingValidator
+
+
+@getter([config_should_not_be_set, config_should_not_be_appended], sender=Project)
+def fs_url_config_validator(**kwargs):
+    if kwargs["key"] == "pootle_fs.fs_url":
+        fs_type = kwargs["instance"].config.get("pootle_fs.fs_type")
+        if not fs_type:
+            raise ValidationError("You cannot set fs url until fs type is set")
```
Comment (unho) [inline pootle/apps/pootle_fs/getters.py]: Nit: `fs_url`, `fs_type`

File: pootle/apps/pootle_fs/getters.py [hunk 3]
```diff
@@ -59,3 +64,16 @@ def fs_url_validator_getter(**kwargs_):
...
+        plugin_class = fs_plugins.gather(Project).get(fs_type)
+        if plugin_class:
```
Comment (unho) [inline pootle/apps/pootle_fs/getters.py]: Shouldn't some error be thrown in case no plugin can be fetch?

File: tests/pootle_fs/utils.py (hunk 1 above)
Comment (phlax) [reply tests/pootle_fs/utils.py]: this test is a bit messy - im gonna clean it up after looking at validation of the fs_type

File: pootle/apps/pootle_fs/getters.py (hunk 3 above)
Comment (phlax) [reply pootle/apps/pootle_fs/getters.py]: see above comment

**Pull request merged**